discovery-infra/download_logs.py

```diff
@@ -54,28 +56,37 @@
 def main():
     args = handle_arguments()
 
+    threads: List[Thread] = []
+
     if args.sosreport:
-        gather_sosreport_data(output_dir=args.dest)
+        threads.append(Thread(target=gather_sosreport_data, kwargs={"output_dir":args.dest}))
 
     client = create_client(url=args.inventory_url, timeout=CONNECTION_TIMEOUT)
     if args.cluster_id:
         cluster = client.cluster_get(args.cluster_id)
-        download_cluster_logs(client, json.loads(json.dumps(cluster.to_dict(), sort_keys=True, default=str)), args.dest,
-                              args.must_gather, args.update_by_events, pull_secret=args.pull_secret)
+        threads.append(Thread(target=download_cluster_logs,
+            kwargs={"client":client, "cluster":json.loads(json.dumps(cluster.to_dict(), sort_keys=True, default=str)), "dest":args.dest,
+                    "update_by_events":args.update_by_events, "must_gather":args.must_gather, "pull_secret":args.pull_secret}))
     else:
         clusters = get_clusters(client, args.download_all)
 
         if not clusters:
             log.info('No clusters were found')
-            return
 
         for cluster in clusters:
             if args.download_all or should_download_logs(cluster):
-                download_cluster_logs(client, cluster, args.dest, args.must_gather, args.update_by_events,
-                                      pull_secret=args.pull_secret)
+                threads.append(Thread(target=download_cluster_logs,
+                    kwargs={"client":client, "cluster": cluster, "dest":args.dest, "update_by_events": args.update_by_events,
+                        "must_gather": args.must_gather, "pull_secret": args.pull_secret}))
 
         log.info("Cluster installation statuses: %s", dict(Counter(cluster["status"] for cluster in clusters).items()))
 
+    for t in threads:
+        t.start()
+
+    for t in threads:
+        t.join()
```

Comment:
Can we do something better with this threads usage?
I would rather have a simple code that takes a bit than a complex one that is more efficient in time

Reply:
Irrelevant to this PR.